_posts/2020-07-12-amiko-hydras-proof.markdown

```diff
@@ -0,0 +1,598 @@
+as the Buchholz function is simpler than all of those, so you might as well jump up to there.
+
+![A scatter plot with connecting lines, with strength of a system on the horizontal axis and complexity of a system on the vertical axis. Each highlighted point is the next stronger system which is minimally more complex. They are, in order, the integers, Cantor normal form, 2-argument Veblen normal form, Buchholz normal form, and Taranovsky's C in the Degrees of Recursive Inaccessibility variant.](/assets/hydras/ordinal-systems.png)
+
+My next major leap was to [Taranovsky's C](https://web.archive.org/web/20200708201005/http://web.mit.edu/dmytro/www/other/OrdinalNotation.htm),
+specifically the "Degrees of Recursive Inaccessibility" system, which is the weakest one on that page.
+It took some reading up on background concepts and examining that system until I was comfortable working in it,
+and it really does not help that that page contains very few examples.
+Once that was done, I was ready to attempt a proof of termination for Amiko hydras, and that proof itself was actually relatively easy.
+
+</div>
+
+# Structure of the proof
+{: #proof-structure}
+
+{% include collapser.markdown %}
+
+<div>
+
+All ordinals are written in Taranovsky's DoRI \\\(C\\\) normal form.
+
+Let \\\(\mathbb{H}\\\) be the set of all hydras.
+Let \\\(0_H = (\star:)\\\).
+Let \\\(\chi\\\) be the limit of Taranovsky's DoRI \\\(C\\\), which is the supremum of all ordinals that can be built using \\\(C\\\) and \\\(0\\\).
+
+Let \\\(P\\\) be "replace all 3-tuples \\\((a, b, c)\\\) with \\\(C(a, b, c)\\\)".
+
+Define \\\(a \lessapprox b\\\) like so:
+
+**1. Definition of termination.**
+\\\(\forall A \in \mathbb{H}, \pi(A) = (A = 0_H \lor \pi(S(A)))\\\)
+Note that \\\(S\\\) is nondeterministic due to the free variable \\\(N\\\), so there is an implied \\\(\forall N \in \mathbb{N}\\\).
+
+*In words, \\\(\pi\\\) is the predicate "this hydra terminates".
+The hydra \\\(0_H\\\) is already considered terminated, and any other hydra terminates if after a step it becomes a hydra which terminates.*
+
+**2. Correspondence from hydras to ordinals.**
+We define a function \\\(M: \mathbb{H} \to \chi\\\) with \\\(M(0_H) = 0\\\)
+We define a function \\\(K\\\) which is like \\\(M\\\) except its output is a tuple, with \\\(M\\\) then defined as \\\(M(A) = P(K(A))\\\).
+Let \\\(\tau(\alpha, m) = (\forall A \in \mathbb{H}, Q(A) \le m \implies ( M(A) = \alpha \implies \pi(A)) )\\\)
+We also fix the definition of hydra comparison as \\\((A \lesseqgtr B) = (M(A) \lesseqgtr M(B))\\\),
+which is needed because the original post on the hydras did not explicitly give a comparison algorithm.
+
+*In words, \\\(\tau\\\) is the predicate "for all hydras which correspond to this ordinal, that hydra terminates".
+There may be no such hydras, in which case it is vacuously true.*
+
+**3. Definition of sub-expression comparison.**
+Detine \\\(a \lessapprox b \iff a \lessapprox^\emptyset b\\\)
```

Comment:
Small spelling mistake: should be "Define".

Reply:
Yep, that's a typo, thanks for spotting that.